test/hotspot/jtreg/runtime/Safepoint/TestAbortVMOnSafepointTimeout.java

```diff
@@ -56,6 +57,31 @@ public static void main(String[] args) throws Exception {
         );
 
         OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        verifyAbortVmApplied(output);
+    }
+
+    public static void testGracePeriodAppliedBeforeVmAbort() throws Exception {
+        ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                "-Xbootclasspath/a:.",
+                "-XX:+UnlockDiagnosticVMOptions",
+                "-XX:+WhiteBoxAPI",
+                "-XX:+SafepointTimeout",
+                "-XX:+SafepointALot",
+                "-XX:+AbortVMOnSafepointTimeout",
+                "-XX:AbortVMOnSafepointTimeoutDelay=1500",
+                "-XX:SafepointTimeoutDelay=50",
+                "-XX:GuaranteedSafepointInterval=1",
+                "-XX:-CreateCoredumpOnCrash",
+                "-Xms64m",
+                "TestAbortVMOnSafepointTimeout$TestWithDelay"
+        );
+
+        OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        output.shouldContain(TestWithDelay.PRE_STALL_TEXT);
+        verifyAbortVmApplied(output);
```

Comment:
Surely in this case there is no abort ??? Otherwise how are you actually testing the delay?

Reply:
@dholmes-ora I've read through the Test and think I now understand it, don't believe it needs a change but maybe I've missed something.

Process is started with 1500ms arming delay and 50ms safepoint timeout, with abort on timeout set.

A 999ms pause is induced. This would trigger the timeout and hence abort, but we are still within the arming delay window. We output PRE_STALL_TEXT so that the test can verify the arming delay succeeded. (tested on line 80).

A further 999ms pause is induced. This will take us past the end of the arming delay and therefore should trigger the VM abort. This is why we still expect the VM to abort (tested line 81).

tl;dr The test is checking that the abort timeout is initially disarmed, and also that it is correctly rearmed, hence why we still get an abort.

Reply:
@pete23hsbc thank you for that explanation. I see now how the test is expecting to work.